hazelcast-jet-core/src/main/java/com/hazelcast/jet/impl/execution/init/PartitionArrangement.java

```diff
@@ -35,62 +33,58 @@
 class PartitionArrangement {
     /**
      * Mapping from each remote member address to the partition IDs it owns.
+     * Members without partitions are missing.
      */
-    final Supplier<Map<Address, int[]>> remotePartitionAssignment;
-    private final Supplier<int[]> localPartitions;
-    private final Supplier<int[]> allPartitions;
+    private final Map<Address, int[]> remotePartitionAssignment;
+
+    /** An array of [0, 1, 2, ... partitionCount-1] */
+    private final int[] allPartitions;
+
+    /** Array of local partitions */
+    private final int[] localPartitions;
 
     PartitionArrangement(Address[] partitionOwners, Address thisAddress) {
-        localPartitions = memoize(() -> arrangeLocalPartitions(partitionOwners, thisAddress));
-        allPartitions = memoize(() -> arrangeAllPartitions(partitionOwners, localPartitions.get()));
-        remotePartitionAssignment = memoize(() -> remotePartitionAssignment(partitionOwners, thisAddress));
+        localPartitions = IntStream.range(0, partitionOwners.length)
+                .filter(partitionId -> thisAddress.equals(partitionOwners[partitionId]))
+                .toArray();
+
+        allPartitions = IntStream.range(0, partitionOwners.length).toArray();
+
+        remotePartitionAssignment = IntStream.range(0, partitionOwners.length)
+                .filter(partitionId -> !thisAddress.equals(partitionOwners[partitionId]))
+                .boxed()
+                .collect(groupingBy(partitionId -> partitionOwners[partitionId],
+                        collectingAndThen(Collectors.toList(), l -> l.stream().mapToInt(i -> i).toArray())));
+    }
+
+    Map<Address, int[]> getRemotePartitionAssignment() {
+        return remotePartitionAssignment;
     }
 
     /**
-     * Determines for each processor instance the partition IDs it will be in charge of
-     * (processors are identified by their index). The method is called separately for
-     * each edge. For a distributed edge, only partitions owned by the local member need
-     * to be assigned; for a non-distributed edge, every partition ID must be assigned.
-     * Local partitions will get the same assignments in both cases, and repeating the
```

Comment:
are we removing this condition? if so, why?

Reply:
Yes, this PR is removing it? I'd reverse the question: why would we keep it? It's not used, it's not documented and it's very rarely beneficial. So the reason is to simplify the code.

Reply:
As discussed, currently we don't make use of this feature anywhere and it makes the code unnecessarily complex.